File: android/app/src/main/kotlin/com/ghoststream/vpn/util/FormatUtils.kt
package com.ghoststream.vpn.util

object FormatUtils {

    fun formatBytes(bytes: Long): String = when {
        bytes < 1_024L              -> "$bytes B"
        bytes < 1_048_576L          -> "%.1f KB".format(bytes / 1_024.0)
        bytes < 1_073_741_824L      -> "%.1f MB".format(bytes / 1_048_576.0)
        else                        -> "%.2f GB".format(bytes / 1_073_741_824.0)
    }

    fun formatSpeed(bytes: Long, elapsedSecs: Long): String {
        if (elapsedSecs <= 0) return "0 B/s"
        return "${formatBytes(bytes / elapsedSecs)}/s"
    }

    fun formatDuration(seconds: Long): String {
        val h = seconds / 3600
        val m = (seconds % 3600) / 60
        val s = seconds % 60
        return "%02d:%02d:%02d".format(h, m, s)
    }
}

File: phantom-vpn-ios/PhantomVPN/Helpers/FormatUtils.swift
import Foundation

enum FormatUtils {
    static func formatBytes(_ bytes: UInt64) -> String {
        switch bytes {
        case 0..<1_024:
            return "\(bytes) B"
        case 1_024..<1_048_576:
            return String(format: "%.1f KB", Double(bytes) / 1_024.0)
        case 1_048_576..<1_073_741_824:
            return String(format: "%.1f MB", Double(bytes) / 1_048_576.0)
        default:
            return String(format: "%.2f GB", Double(bytes) / 1_073_741_824.0)
        }
    }

    static func formatDuration(seconds: Int) -> String {
        let h = seconds / 3600
        let m = (seconds % 3600) / 60
        let s = seconds % 60
        return String(format: "%02d:%02d:%02d", h, m, s)
    }
}
